tests/recipe_tests/test_HDFReferenceRecipe.py

```diff
@@ -73,3 +73,44 @@ def test_multi(netcdf_local_file_pattern_sequential, tmpdir, with_intake):
         )
         ds = xr.open_dataset(m, engine="zarr", chunks={}, consolidated=False)
     assert (ds.foo == expected.foo).all()
+
+
+def _drop_lon(ds):
+    return ds.drop_vars(["lon"])
+
+
+@pytest.mark.parametrize("process", [None, _drop_lon])
+@pytest.mark.parametrize("with_intake", [True, False])
+def test_process(netcdf_local_file_pattern_sequential, tmpdir, with_intake, process):
+    file_pattern = netcdf_local_file_pattern_sequential
+    paths = [f for _, f in list(file_pattern.items())]
+    expected = xr.open_mfdataset(paths, engine="h5netcdf")
+    if process:
+        expected = process(expected)
+
+    recipe = HDFReferenceRecipe(file_pattern, process_input=process)
+
+    # make sure assigning storage later works
+    out_target = FSSpecTarget(fs=fsspec.filesystem("file"), root_path=str(tmpdir))
+    metadata_cache = MetadataTarget(fs=fsspec.filesystem("file"), root_path=tempfile.mkdtemp())
+    recipe.target = out_target
+    recipe.metadata_cache = metadata_cache
+
+    recipe.to_dask().compute(scheduler="sync")
+
+    assert out_target.exists("reference.json")
+    assert out_target.exists("reference.yaml")
+
+    if with_intake:
+        cat = intake.open_catalog(out_target._full_path("reference.yaml"))
+        ds = cat.data.read()
+    else:
+        m = fsspec.get_mapper(
+            "reference://",
+            fo=out_target._full_path("reference.json"),
+            target_protocol="file",
+            remote_protocol="file",
+            skip_instance_cache=True,
+        )
+        ds = xr.open_dataset(m, engine="zarr", chunks={}, consolidated=False)
+    xr.testing.assert_equal(ds, expected)
```

Comment:
Instead of creating a new test by copy-pasting the previous one, let's add the `@pytest.mark.parametrize("process", [None, _drop_lon])` decorator (and associated logic) to `test_multi`. This would be a lot more compact and would help keep our test suite running more quickly (fewer tests run total). I don't see what we gain by running this test with `process=None`.

Reply:
Sounds good to me!